Plots/XY/NCL_xy_18.py

```diff
@@ -0,0 +1,124 @@
+- Maximizing plots after they've been created
+
+See the [original NCL example](https://www.ncl.ucar.edu/Applications/Scripts/xy_18.ncl)
+"""
+
+###############################################################################
+# Basic imports
+import numpy as np
+import xarray as xr
+from matplotlib import pyplot as plt
+import matplotlib.ticker as tic
+
+###############################################################################
+# Open files and read in monthly data
+
+def assume_noleap_calendar(ds):
+    ds.time.attrs['calendar'] = 'noleap'
+    return xr.decode_cf(ds)
+
+nfiles = ["../../data/netcdf_files/TREFHT.B06.66.atm.1890-1999ANN.nc",
+          "../../data/netcdf_files/TREFHT.B06.67.atm.1890-1999ANN.nc",
+          "../../data/netcdf_files/TREFHT.B06.68.atm.1890-1999ANN.nc",
+          "../../data/netcdf_files/TREFHT.B06.69.atm.1890-1999ANN.nc"]
+nds = xr.open_mfdataset(nfiles, concat_dim='case', combine='nested',
+                        preprocess=assume_noleap_calendar, decode_times=False)
+
+vfiles = ["../../data/netcdf_files/TREFHT.B06.61.atm.1890-1999ANN.nc",
+          "../../data/netcdf_files/TREFHT.B06.59.atm.1890-1999ANN.nc",
+          "../../data/netcdf_files/TREFHT.B06.60.atm.1890-1999ANN.nc",
+          "../../data/netcdf_files/TREFHT.B06.57.atm.1890-1999ANN.nc"]
+vds = xr.open_mfdataset(vfiles, concat_dim='case', combine='nested',
+                        preprocess=assume_noleap_calendar, decode_times=False)
+
+gds = xr.open_dataset("../../data/netcdf_files/gw.nc")
+gds = gds.expand_dims(dim={'lon': nds.lon})
+
+###############################################################################
+# Some parameters
+nyrs = len(nds["time"])
+nlon = len(nds["lon"])
+nlat = len(nds["lat"])
+time =
+
+###############################################################################
+# OBS
+
+obs = np.loadtxt("../../data/ascii_files/jones_glob_ann_2002.asc", dtype=float)
+
+###############################################################################
+# NCL-based Weighted Mean Function
+
+def horizontal_weighted_mean(var, wgts):
+    return (var * wgts).sum(dim=['lat', 'lon']) / wgts.sum(dim=['lat', 'lon'])
+
+###############################################################################
+# NATURAL
+
+gavn = horizontal_weighted_mean(nds["TREFHT"], gds["gw"])
+gavan = gavn - gavn.isel(time=slice(0,30)).mean(dim='time')
+
+###############################################################################
+# ALL
+
+gavv = horizontal_weighted_mean(vds["TREFHT"], gds["gw"])
+gavav = gavv - gavv.isel(time=slice(0,30)).mean(dim='time')
+
+###############################################################################
+# CALCULATE ENSEMBLE MIN & MAX
+
+gavan_min = gavan.min(dim='case')
+gavan_max = gavan.max(dim='case')
+gavan_avg = gavan.mean(dim='case')
+
+gavav_min = gavav.min(dim='case')
+gavav_max = gavav.max(dim='case')
+gavav_avg = gavav.mean(dim='case')
+
+obs_avg = obs[34:144] - np.mean(obs[34:64])
```

Comment:
Same here, why 34,144; and 34, 64

Reply:
This is what was done in the NCL example.  And the data is from just a single-column text file, so there is no metadata.  Would you recommend that I manually *add* metadata and create an Xarray `Dataset`?

Reply:
Done.